src/ImageSharp/Image.WrapMemory.cs

```diff
@@ -150,5 +150,70 @@ public abstract partial class Image
             int height)
             where TPixel : unmanaged, IPixel<TPixel>
             => WrapMemory(Configuration.Default, pixelMemoryOwner, width, height);
+
+        /// <summary>
+        /// Wraps an existing contiguous memory area of 'width' x 'height' pixels,
+        /// allowing to view/manipulate it as an ImageSharp <see cref="Image{TPixel}"/> instance.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel type</typeparam>
+        /// <param name="configuration">The <see cref="Configuration"/></param>
+        /// <param name="byteMemory">The byte memory representing the pixel data.</param>
+        /// <param name="width">The width of the memory image.</param>
+        /// <param name="height">The height of the memory image.</param>
+        /// <param name="metadata">The <see cref="ImageMetadata"/>.</param>
+        /// <exception cref="ArgumentNullException">The configuration is null.</exception>
+        /// <exception cref="ArgumentNullException">The metadata is null.</exception>
+        /// <returns>An <see cref="Image{TPixel}"/> instance</returns>
+        public static Image<TPixel> WrapMemory<TPixel>(
+            Configuration configuration,
+            Memory<byte> byteMemory,
+            int width,
+            int height,
```

Comment:
How do we validate, what do we throw if the sizes mismatch? Also needs tests for that case.

Reply:
I'm really just passing the new `Memory<TPixel>` instance to the `Image<TPixel>` constructor like the previously existing APIs, so I just assumed `Image<TPixel>` would perform the input validation already? My idea here was to avoid repeating the same tests if not needed (eg. the overloads taking a `Memory<TPixel>` argument have no validation either).

Let me know though if you want any changes being done here (possibly to previous overloads too then?).

Reply:
I can't find any validation in the current codebase that ensures the memory size matches. We check in `MemoryGroup<T>.Wrap(...)` that the buffers are uniform but there doesn't seem to be anything else.

Reply:
Wait hold on, doesn't that mean the already existing APIs right now lack validation too then? 🤔

I mean, like this:

https://github.com/SixLabors/ImageSharp/blob/85f91f6992da966dc10216b664e3ff833f425e56/src/ImageSharp/Image.WrapMemory.cs#L98-L111

Reply:
Yeah, I think so. I could be wrong but I couldn't find anything. A unit test would quickly tell us.

Reply:
Yeah, looks like at the end of the day the wrapped memory just gets passed here:

https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Memory/Buffer2D%7BT%7D.cs#L30-L40

And the constructor doesn't validate that the input group actually matches the specified size.
Will add a new unit test and will add the checks to both the new and previous APIs then! 👍

Reply:
That matches what I found. Thanks!

Reply:
Done in https://github.com/SixLabors/ImageSharp/pull/1314/commits/fa59a73c39d6ea25edfb690957be2c494176893f.